internal/kafka/internal/handlers/cluster.go

```diff
@@ -173,3 +173,34 @@ func (h clusterHandler) DeregisterEnterpriseCluster(w http.ResponseWriter, r *ht
 	}
 	handlers.HandleDelete(w, r, cfg, http.StatusAccepted)
 }
+
+func (h clusterHandler) Get(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			ValidateKafkaClaims(ctx, ValidateOrganisationId()),
+		},
+		Action: func() (i interface{}, serviceError *errors.ServiceError) {
+			// error checked in the validate, no need to check again
+			claims, _ := getClaims(ctx)
+			orgId, _ := claims.GetOrgId()
+
+			if !claims.IsOrgAdmin() {
+				return nil, errors.New(errors.ErrorUnauthorized, "non admin user not authorized to perform this action")
+			}
+
+			id := mux.Vars(r)["id"]
+			cluster, err := h.clusterService.FindClusterByID(id)
+			if err != nil {
+				return nil, err
+			}
+
+			if cluster == nil || cluster.OrganizationID != orgId || cluster.ClusterType != api.EnterpriseDataPlaneClusterType.String() {
+				return nil, errors.NotFound("enterprise data plane cluster with id='%v' not found within organization: %s", id, orgId)
+			}
+
+			return presenters.PresentEnterpriseCluster(*cluster), nil
```

Comment:
I am wondering if this endpoint could return the fleetshard addon parameters as well? 

Alternatively, we could have a split like what the connectors team have

```
GET /api/kafkas_mgmt/v1/clusters/{id} - all users of the org can see the cluster and no fleetshard parameters are returned
GET /api/kafkas_mgmt/v1/clusters/{id}/addon_parameters - only admin users can access this endpoint and it returns the addon parameters
```

The advantage is that this endpoint can be used to offer a client side reconciliation of the addon parameters post registration.

Reply:
What do you think? @pawelpaszki